src/utils.rs

```diff
@@ -1,7 +1,7 @@
 use std::sync::Once;
 
 use bao::Hash;
-use bech32::{decode, encode, FromBase32, ToBase32, Variant};
+use bech32::{decode, encode, Bech32m, Hrp};
```

Comment:
I will need to review the bao::Hask to understand the minus of "FromBase32, etc.and what is Hrp"

Reply:
This is for bech32 encoding. HRP stands for Human Readable Part. You can read more in the BIP:
https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki